ouroboros-network/src/Ouroboros/Network/Mux.hs

```diff
@@ -83,27 +79,28 @@ muxControl pmss md = do
         atomically $ putTMVar w blob
         atomically $ writeTBQueue (tsrQueue pmss) (TLSRDemand Muxcontrol md (Wanton w))
 
--- | muxDuplex creates a duplex channel for a specific 'MiniProtocolId' and 'MiniProtocolMode'.
-muxDuplex :: (MonadSTM m, MonadSay m) =>
+-- | muxChannel creates a duplex channel for a specific 'MiniProtocolId' and 'MiniProtocolMode'.
+muxChannel :: (MonadSTM m, MonadSay m) =>
     PerMuxSharedState m ->
     MiniProtocolId ->
     MiniProtocolMode ->
     TMVar m BL.ByteString ->
-    Duplex m m CBOR.Encoding BS.ByteString
-muxDuplex pmss mid md w = uniformDuplex snd_ rcv
+    Channel m BL.ByteString
+muxChannel pmss mid md w =
+    Channel {send, recv}
   where
-    snd_ encoding = do
+    send encoding = do
         -- We send CBOR encoded messages by encoding them into by ByteString
         -- forwarding them to the 'mux' thread, see 'Desired servicing semantics'.
         --say $ printf "send mid %s mode %s" (show mid) (show md)
```

Comment:
I think we can remove the commented out `say ...`

Reply:
I agree, though perhaps can be improved independently of this PR.

Reply:
That's ok.